### ISSUE TITLE
[Inquiry] Render method

### ISSUE THREAD

**Issue by MysticAura** [Chisel-Team/Chisel#181]
Are you guys using a special renderer for your blocks? 

I'm asking in hopes that there is a way for MultiStorage mod to be able to clone your blocks with its chameleon storage. Currently it only shows missing texture when he clones the rendered block model.

Any ideas you could give would be much appreciated.

**Comment by Drullkus** [comment]
Paging @minecreatr and @tterrag1098 

Alternatively, @lordmau5 could help

**Comment by tterrag1098** [comment]
Our model requires the extended state to be provided. However, with no extended state given it should still render the default block. I would need to see how exactly you are "cloning" our render to see what the problem is.

**Comment by AtomicBlom** [comment]
I've been able to render their blocks with little difficulty using a default state.

**Comment by ShetiPhian** [comment]
I'll chime in here,
While the chameleon block is able to clone most blocks it doesn't work with everything, MysticAura asked me to support Chisel, but since I was already using the extended state I figured it was using the TESR.

The block's getExtendedState, embeds the extended state of the cloned block into its own extended state.
In its IBakedModel.getQuads it extracts the embedded iblockstate, gets the registered model for it, and returns the getQuads from that model.

But, while typing out the code I noticed an overlooked simple mistake, I forgot to clean the cloned extended state before retrieving the model, causing the missing model to be returned.
I've looked it over a few times, hell yesterday I even edited the line directly bellow it, but it wasn't until typing out the code here did I notice the error.

**tl;dr**
I overlooked the missing IExtendedBlockState.getClean() before calling getModelForState
